src/DynamicData/Cache/ObservableCache.cs

```diff
@@ -240,4 +322,103 @@ private void InvokePreview(ChangeSet<TObject, TKey> changes)
             }
         }
     }
+
+    private void InvokeCountNext()
+    {
+        lock (_locker)
+        {
+            if (_countChanged.IsValueCreated)
+            {
+                _countChanged.Value.OnNext(_readerWriter.Count);
+            }
+        }
+    }
+
+    private void ResumeCount()
+    {
+        lock (_locker)
+        {
+            Debug.Assert(_suspensionTracker.IsValueCreated, "Should not be Resuming Count without Suspend Count instance");
+            _suspensionTracker.Value.ResumeCount();
+        }
+    }
+
+    private void ResumeNotifications()
+    {
+        lock (_locker)
+        {
+            Debug.Assert(_suspensionTracker.IsValueCreated, "Should not be Resuming Notifications without Suspend Count instance");
+            _suspensionTracker.Value.ResumeNotifications();
+        }
+    }
+
+    private class SuspensionTracker(Action<ChangeSet<TObject, TKey>> onResumeNotifications, Action onResumeCount) : IDisposable
+    {
+        private readonly BehaviorSubject<bool> _notifySuspendSubject = new(false);
+
+        private readonly Action<ChangeSet<TObject, TKey>> _onResumeNotifications = onResumeNotifications;
+
+        private readonly Action _onResumeCount = onResumeCount;
+
+        private IEnumerable<Change<TObject, TKey>> _pending = Enumerable.Empty<Change<TObject, TKey>>();
+
+        private int _countSuspendCount;
+
+        private int _notifySuspendCount;
+
+        public bool CountSuspended => _countSuspendCount > 0;
+
+        public bool NotificationsSuspended => _notifySuspendCount > 0;
+
+        public IObservable<bool> NotificationsSuspendedStatus => _notifySuspendSubject;
+
+        public void AddPending(IEnumerable<Change<TObject, TKey>> changes)
+        {
+            Debug.Assert(changes is not null, "Don't pass in a null Enumerable");
+            Debug.Assert(NotificationsSuspended, "Shouldn't be adding pending values if notifications aren't suspended");
+            _pending = _pending.Concat(changes);
```

Comment:
Surely a `List<T>` that gets reused would be better than chained enumerables?

Reply:
I'm not sure.  I thought this would be better by avoiding the copy to the list.  The ChangeSet is already made and allocated.  Putting them in a list would definitely be two copies.  This is just one (when it makes the final ChangeSet).

Reply:
So, ultimately we're trading some extra allocations on each suspended set of changes, in favor of skipping an iteration over them, to copy them into a buffer. I'm not sure which would be more "efficient" overall. Sounds like it's worth a benchmark.